lib/ice_nine/support/recursion_guard.rb

```diff
@@ -31,8 +33,8 @@ def initialize
       # @return [Object]
       def guard(object)
         caller_object_id = object.__id__
-        return object if @object_ids.key?(caller_object_id)
-        @object_ids[caller_object_id] = nil
+        return object if @object_ids.member?(caller_object_id)
+        @object_ids.add caller_object_id
```

Comment:
I think we generally use explicit parens on method calls that are inside `def` and `defs` nodes. This project was as far as I recall, not yet ported to `devtools` that would enforce this via rubocop.

`ice_nine` is much older than most of our infrastructure.